File: frontend/src/types/manufacturer.ts
export interface Manufacturer {
  manufacturer_id: number;
  manufacturer_name: string;
  country_code?: string;
  license_number?: string;
  address?: string;
  phone?: string;
  email?: string;
  status?: string;
  created_at?: string;
  updated_at?: string;
}

export interface ManufacturerWithBrands extends Manufacturer {
  brands?: BrandSummary[];
}

export interface BrandSummary {
  brand_id: number;
  brand_name: string;
  generic_name?: string;
}

export interface ManufacturerCreate {
  manufacturer_name: string;
  country_code?: string;
  license_number?: string;
  address?: string;
  phone?: string;
  email?: string;
  status?: string;
}

export interface ManufacturerUpdate {
  manufacturer_name?: string;
  country_code?: string;
  license_number?: string;
  address?: string;
  phone?: string;
  email?: string;
  status?: string;
}

File: app/models/manufacturer.py
from sqlalchemy import Column, Integer, String, DateTime, Enum
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.db.base import Base
from app.core.constants import ManufacturerStatus


class Manufacturer(Base):
    __tablename__ = "manufacturers"
    
    manufacturer_id = Column(Integer, primary_key=True, index=True)
    manufacturer_name = Column(String(255), nullable=False)
    country_code = Column(String(2))  # ISO country code
    license_number = Column(String(50))
    address = Column(String(500))
    phone = Column(String(50))
    email = Column(String(255))
    status = Column(Enum(ManufacturerStatus), default=ManufacturerStatus.ACTIVE)
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), onupdate=func.now())
    
    # Relationship to brand names
    brand_names = relationship("BrandName", back_populates="manufacturer")
    
    def __repr__(self):
        return f"<Manufacturer(manufacturer_name='{self.manufacturer_name}', status='{self.status}')>"
